libcontainer/init_linux.go

```diff
@@ -499,6 +499,26 @@ func signalAllProcesses(m cgroups.Manager, s os.Signal) error {
 	if err := m.Freeze(configs.Frozen); err != nil {
 		logrus.Warn(err)
 	}
+	// If signal is SIGKILL try manager.Kill()
+	// inorder to leverage "cgroup.kill" instead
+	// of iterating over each process. "cgroup.kill"
+	// was added in Linux kernel 5.14 onwards read
+	// more about it https://lwn.net/Articles/855924/
+	if s == unix.SIGKILL {
+		// If manager.Kill() failed
+		// fallback to legacy method
+		err := m.Kill()
+		// If err is nil
+		// Means "cgroup.kill" was successful
+		// return early from function
+		if err == nil {
+			if err := m.Freeze(configs.Thawed); err != nil {
+				logrus.Warn(err)
```

Comment:
logrus.WithError(err).Warnf(“some human-readable message here”

Reply:
Sure :)

Reply:
Freeze is removed in latest commit. Please check https://github.com/opencontainers/runc/pull/3199#discussion_r703047099